kernel-rs/src/pipe.rs

```diff
@@ -139,22 +136,23 @@ impl AllocatedPipe {
             read_waitchannel: WaitChannel::new(),
             write_waitchannel: WaitChannel::new(),
         };
-
-        // FIXME: You cannot get the mutable reference of Rc.
-        (*f0).typ = Filetype::PIPE;
-        (*f0).readable = true;
-        (*f0).writable = false;
-        (*f0).pipe = Self { ptr };
-        (*f1).typ = Filetype::PIPE;
-        (*f1).readable = false;
-        (*f1).writable = true;
-        (*f1).pipe = Self { ptr };
+        let mut f0 = RcFile::alloc(true, false).ok_or_else(|| {
```

Comment:
Rust에서 아무 것도 리턴하지 않는 것은 `()` 를 리턴하는 것과 같습니다. `kfree` 가 그런 함수이므로,
```suggestion
        let mut f0 = RcFile::alloc(true, false).ok_or_else(|| kfree(ptr as _))?;
```

Reply:
https://github.com/kaist-cp/rv6/pull/211/commits/012fc469f533ba8197c5ecb8e58c47035701d442